### DISCUSSION TITLE
[Linux] Are png writes and loads locked to a single thread?

### DISCUSSION THREAD

**Discussion by DejanPelzel** [dlemstra/Magick.NET#623] [General] [answered]
Hello,

We recently noticed some drag in image processing and it looks like png Write/Load calls are blocking to a single thread on Ubuntu/Linux. If I understand correctly that might be caused by libpng (version 1.6)?

Is there any way to work around that? We're doing heavy multi-threading and this is really slowing things down.

Thanks!

**Comment by dlemstra** [comment]
It does look like the png coder is single threaded. It seems that `IMPNG_SETJMP_IS_THREAD_SAFE` is not set in the Linux/MacOS build. Will need to figure out why this is not set. Will get back to you when I have more information.

**Comment by dlemstra** [comment, marked as answer]
We just pushed a patch to ImageMagick to fix this. I can probably publish a new release this week to resolve this for you.

**Reply by DejanPelzel** [reply]
That would be great, thank you so much!

**Reply by dlemstra** [reply]
The new version is now being pushed to nuget. Should be available within an hour.
